### PR TITLE
Feature/add logout

### PR DESCRIPTION
**Pull request by flibbertigibbet**

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -21,6 +22,10 @@
     <string name="error_login_unknown">Unknown error logging in.</string>
     <string name="error_login_network">Network error logging in.</string>
 
+    <!-- logout -->
+    <string name="logout_dialog_title">Are you sure you want to log out?</string>
+    <string name="logout_dialog_message">You will need to log in again to enter or manage records after logging out. Logging in will require Internet access. Are you sure you want to log out now?</string>
```
Comment (ddohler) [inline app/src/main/res/values/strings.xml:27]: One of the source comments above implies that records which have not been uploaded to the server will be lost when logging out; if that's the case then I think this warning could be made a bit clearer.

**Comment by ddohler** [comment]
+1 -- do you know what's up with Travis?

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (flibbertigibbet) [inline app/src/main/res/values/strings.xml:27]: No, logging out does not affect the local database of records. They'll stay there until the user logs in again, under any account. The logout warning prompt is there to make sure they don't accidentally log out while in the field without Internet access, leaving them unable to use the app again until they get back on a network somewhere.

It is necessary to upload/clear the local records before updating the schema; I take it that's the other warning message you're looking at. There's no risk of record loss with the update. The old records need to be uploaded before the schema changes in order to deal with the class reloading cleanly.

**Comment by flibbertigibbet** [comment]
I'm pretty sure Travis is grumpy because it is missing the SSO configuration, and I moved around some connection things to happen on activity start/stop. Tests pass locally. I'm wrapping some SSO stuff in error handling for Travis, which will probably be good to have anyways.

**Pull request merged**